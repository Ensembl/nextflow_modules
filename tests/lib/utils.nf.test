
nextflow_function {

    name "Test Functions"
    script "lib/utils.nf"

	

    test("Test Function get_key_list") {

        function "get_key_list"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function read_json") {

        function "read_json"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function parse_list_param") {

        function "parse_list_param"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function generate_url") {

        function "generate_url"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function extract_url_args") {

        function "extract_url_args"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function pattern = ~/mysql:\/\/(.*?)(:(.*?)?)?@(.*?):(\d+)\/?") {

        function "pattern = ~/mysql:\/\/(.*?)(:(.*?)?)?@(.*?):(\d+)\/?"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function url_parts = url_string.tokenize") {

        function "url_parts = url_string.tokenize"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function user_info = url_parts[1].split") {

        function "user_info = url_parts[1].split"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function (host, port) = url_parts[2].split") {

        function "(host, port) = url_parts[2].split"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function database = url_parts.size") {

        function "database = url_parts.size"

        when {
            function {
                """
                // define inputs of the function here. Example:
                // input[0] = 1
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

}