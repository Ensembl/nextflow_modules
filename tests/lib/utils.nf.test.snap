{
    "Parse_map_param with run all": {
        "content": [
            [
                "test",
                "lorem"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T12:28:26.443400428"
    },
    "Test Function extract_url_args": {
        "content": [
            {
                "user": "ensro",
                "password": null,
                "host": "mysql-eg-test-prod",
                "port": "0000",
                "database": null
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T11:07:26.500484697"
    },
    "Test Function generate_url": {
        "content": [
            "https://ensro@mysql:0"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T10:59:52.815350874"
    },
    "Test Function get_key_list": {
        "content": [
            "['Key1','key2']"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-27T17:33:46.016044367"
    },
    "Test Function read_json": {
        "content": [
            {
                "Key": "value"
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-27T17:33:55.082623428"
    },
    "Test Function parse_list_param": {
        "content": [
            [
                "test"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T10:35:22.070018212"
    },
    "Failure parse_map_param": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T11:41:24.859934595"
    },
    "Failure parse_map_param no value ": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T12:01:59.363815572"
    },
    "Test Function generate_url with password and database": {
        "content": [
            "https://ensro:test@mysql:0/test_database"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T16:49:48.297017432"
    },
    "Failure parse_map_param value not in allowed allowed_values ": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T11:54:20.475570355"
    },
    "Parse_map_param with no value ": {
        "content": [
            "No value specified null"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T12:10:54.223286963"
    },
    "Failure extract_url_args wrong pattern": {
        "content": null,
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-28T16:57:36.491445174"
    }
}